packages/application-shell/src/components/application-shell/application-shell.js

```diff
@@ -368,23 +368,42 @@ RestrictedApplication.propTypes = {
   DEV_ONLY__loadNavbarMenuConfig: PropTypes.func,
 };
 
-const isBooleanAsString = environmentValue =>
-  environmentValue === 'true' || environmentValue === 'false';
+/**
+ * NOTE:
+ *   This function try-catches around a `JSON.parse` and return
+ *   the parsed value whenever possible.
+ *
+ *   This allows parsing most primitive values.
+ *
+ *   - `JSON.parse('null')` => `null`
+ *   - `JSON.parse('1')` => `1`
+ *   - `JSON.parse('["a", "b"]')` => `['a', 'b']`
+ */
+const getIfJSONValue = environmentValueAsString => {
+  let parsedEnvironmentValue;
+
+  try {
+    parsedEnvironmentValue = JSON.parse(environmentValueAsString);
+  } catch (e) {
+    return false;
+  }
+
+  return parsedEnvironmentValue;
+};
```

Comment:
@emmenko this is the easiest way I could think of solving this problem of String encoded primitives.